frontend/openchat-agent/src/services/user/mappers.ts

```diff
@@ -867,40 +951,18 @@ function chatSummary(candid: ApiChatSummary): ChatSummary {
     throw new UnsupportedValueError("Unexpected ApiChatSummary type received", candid);
 }
 
-function chatMetrics(candid: ApiChatMetrics): ChatMetrics {
-    return {
-        audioMessages: Number(candid.audio_messages),
-        cyclesMessages: Number(candid.cycles_messages),
-        edits: Number(candid.edits),
-        icpMessages: Number(candid.icp_messages),
-        giphyMessages: Number(candid.giphy_messages),
-        deletedMessages: Number(candid.deleted_messages),
-        reportedMessages: Number(candid.reported_messages),
-        fileMessages: Number(candid.file_messages),
-        pollVotes: Number(candid.poll_votes),
-        textMessages: Number(candid.text_messages),
-        imageMessages: Number(candid.image_messages),
-        replies: Number(candid.replies),
-        videoMessages: Number(candid.video_messages),
-        polls: Number(candid.polls),
-        reactions: Number(candid.reactions),
-    };
-}
-
-function groupChatSummary(candid: ApiGroupChatSummary): GroupChatSummary {
+function groupChatSummary(candid: ApiGroupChatSummary, limitReadByMeUpTo = true): GroupChatSummary {
```

Comment:
If `limitReadByMeUpTo` is true, then `readByMeUpTo` is capped to the latest message index.
This is usually what we want, but in the case where we load cached groups from the user canister, the cached latest message can be old and so can be a lower index than `readUpToByMe`.